lib/code_fund_web/controllers/api/ad_serve_controller.ex

```diff
@@ -30,6 +30,15 @@ defmodule CodeFundWeb.API.AdServeController do
     end
   end
 
+  defp get_template_slug(property_id, requested_template_slug) do
```

Comment:
Functions which look up objects from the database don't belong in the controller. If it were me, I'd move something like this as a public function in Templates, join to properties where property.id == property_id and I'd rewrite the case as a pattern matched function clause def that the slug.

```
defmodule Templates do
  def for_property_id(property_id, requested_template_slug) do
     from(t in Template, join: p in Property on: p.template_id == t.id,
     where: p.id == ^property_id)
     |> Repo.one
     |> return_template(requested_template_slug)  
   end
  defp return_template(%Template{slug}}, requested_template_slug), when not(is_nil(slug)), do: slug
  defp return_template(_,_), do: requested_template_slug
end

Reply:
This is great feedback. Thanks for pointing me in the right direction.

Reply:
You bet